sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_decorator.py

```diff
@@ -227,33 +235,39 @@ def _validate_args(func, args, kwargs, non_pipeline_inputs):
             any(not isinstance(param, str) for param in non_pipeline_inputs):
         raise UnExpectedNonPipelineParameterTypeError()
     # Positional arguments validate
+    is_support_variable_params = is_private_preview_enabled()
     all_parameters = [param for _, param in signature(func).parameters.items()]
-    # Implicit parameter are *args and **kwargs
-    if any(param.kind in {param.VAR_KEYWORD, param.VAR_POSITIONAL} for param in all_parameters):
-        raise UnsupportedParameterKindError(func.__name__)
-
     all_parameter_keys = [param.name for param in all_parameters]
     non_pipeline_inputs = non_pipeline_inputs or []
     unexpected_non_pipeline_inputs = [param for param in non_pipeline_inputs if param not in all_parameter_keys]
     if unexpected_non_pipeline_inputs:
         raise NonExistParamValueError(func.__name__, unexpected_non_pipeline_inputs)
 
-    empty_parameters = {param.name: param for param in all_parameters if param.default is Parameter.empty}
-    min_num = len(empty_parameters)
-    max_num = len(all_parameters)
-    if len(args) > max_num:
-        raise TooManyPositionalArgsError(func.__name__, min_num, max_num, len(args))
-
-    provided_args = OrderedDict({param.name: args[idx] for idx, param in enumerate(all_parameters) if idx < len(args)})
+    named_parameters = [param for param in all_parameters if param.kind not in [param.VAR_KEYWORD, param.VAR_POSITIONAL]]
+    empty_parameters = {param.name: param for param in named_parameters if param.default is Parameter.empty}
+    # Implicit parameter are *args and **kwargs
+    if not is_support_variable_params:
+        if len(all_parameters) != len(named_parameters):
+            raise UnsupportedParameterKindError(func.__name__)
+
+        min_num = len(empty_parameters)
+        max_num = len(named_parameters)
+        if len(args) > max_num:
+            raise TooManyPositionalArgsError(func.__name__, min_num, max_num, len(args))
+
+    func_args, provided_args = list(args), OrderedDict({})
+    provided_kwargs = OrderedDict({param.name: func_args.pop(0) for param in named_parameters if len(func_args) > 0})
+    for index, arg in enumerate(func_args):
+        provided_args[f"args_{index}"] = func_args[index]
```

Comment:
do we need to hard code args_?

Reply:
you could refer to 1.5 _validate_args, just copy the whole `_validate_args`, it should support variable parameter, I think we don't need args_index as key, just get the param from function.

Reply:
Yes, it's possible to support variable parameters by removing some validations refer to v1.5 _validate_args.
But these variable parameters are also pipeline inputs and still need a key as the input name

Reply:
I will change the hard code args_ to {variable_args_name}_{index}

Reply:
change the parameter name to {variable_args_name}_{index}